src/gui/components/ChannelBoot.tsx

```diff
@@ -322,8 +320,21 @@ const terminalFlow = ({
                         reject('invalid key');
                         return;
                     }
+                    const [channelId, hostVersionHash] = input.split('==');
+                    if (!hostVersionHash) {
+                        reject('invalid key');
+                        return;
+                    }
+                    const clientVersionHash = md5(
+                        getVersionStringFromConfig(socketConfig),
+                    ).slice(0, 4);
+                    if (hostVersionHash !== clientVersionHash) {
+                        reject('client app version incompatible with host');
+                        return;
+                    }
+
                     client
-                        .joinChannel(input)
+                        .joinChannel(channelId + '==')
```

Comment:
🤔 hmm that looks sus ... 

maybe I've misunderstood, but are you depending on the base64 value to end with `==` ... because `=` is a padding char and so not all base64 encoded values end in two or even any equals.

```
❯ echo 'bbbb' | base64
YmJiYgo=

❯ echo 'xx' | base64
eHgK
```

Reply:
ah I thought it did, nicely spotted. I'll fix that